sdk/containerregistry/azure-containerregistry/azure/containerregistry/_models.py

```diff
@@ -92,6 +88,11 @@ def __init__(self, **kwargs):
         self.can_read = kwargs.get("can_read")
         self.can_list = kwargs.get("can_list")
         self.can_write = kwargs.get("can_write")
+        try:
+            self._architecture = ArtifactArchitecture(self._architecture)
```

Comment:
What if self._architecture = ArtifactArchitecture(self._architecture) raises?

Do we want to continue to try self._architecture = ArtifactArchitecture(self._architecture)?

Reply:
Updated to try on "architecture" and "operating_system" separately.